proposals/readonly-ref.md

```diff
@@ -179,16 +179,10 @@ However, since there is no backward compatibility considerations and there are w
 
 ## Metadata representaion.
 
-CLR provides several mechanisms for augmenting signatures - custom attributes, pseudo custom attributes and custom modifiers (modreq/modopt).   
-Either would work, in theory, but we need to decide on one or another.
+We're using a combination of attributes and modifiers on signatures:
 
-Historically `out` uses pseudo custom attribute `[out]`.
-There is a corresponding `[in]` pseudo custom attribute, that could be used.
-
-Attributes can be more attractive for parameters if it is a goal to make a change from ordinary `ref`
-parameter to a `readonly ref` one  nonbreaking to existing callers (in binary, no recompile sense).
-
-`modreq` could be more suitable for `readonly ref` returns if we want to block misuse by the downlevel callers.
+1) `IsReadOnlyAttribute` is added to the framework, to be used on parameters and return types, to indicate that they are `ref readonly`. For older frameworks, the compiler will generate an embedded (hidden) attribute with the same name into the module being built, and we will have a mechanism for disallowing using this attribute in source code.
+2) We will use a `modreq` with the framework type `IsConst`, to prevent other compilers and older versions of the C# compiler to read such signatures.
```

Comment:
Attribute is mandatory and is the marker.

modreq is just to prevent downlevel misuse and in some cases (like: parameters of nonvirtual methods) may not be emitted, since downlevel use is safe.